app/src/db/models/tables/objectModel.js

```diff
@@ -156,6 +159,7 @@ class ObjectModel extends Timestamps(Model) {
       hasPermission(query, userId, permCode) {
         if (userId && permCode) {
           query
+            .withGraphFetched('[objectPermission, bucketPermission]', { joinOperation: 'fullOuterJoinRelated' })
```

Comment:
Optional: Might be worth putting in a comment here to explain the nuance of what this graph fetch is doing. There is probably a significant amount of subtle behavior happening here with the extra joinOperation option.